plan/task/pull.go

```diff
@@ -27,19 +29,42 @@ func (c *pullTask) Run(ctx context.Context, pctx *plancontext.Context, s *solver
 		return nil, err
 	}
 
+	// Extract registry target from source
+	target, err := solver.ParseAuthHost(rawRef)
+	if err != nil {
+		return nil, err
+	}
+
 	// Read auth info
 	if auth := v.Lookup("auth"); auth.Exists() {
 		a, err := decodeAuthValue(pctx, auth)
 		if err != nil {
 			return nil, err
 		}
-		// Extract registry target from source
-		target, err := solver.ParseAuthHost(rawRef)
-		if err != nil {
-			return nil, err
-		}
+
 		s.AddCredentials(target, a.Username, a.Secret.PlainText())
 		lg.Debug().Str("target", target).Msg("add target credentials")
+	} else if target == "docker.io" {
+		// Collect DOCKERHUB_AUTH_USER && DOCKERHUB_AUTH_PASSWORD env vars
+		username, secret := "", ""
+		for _, envVar := range os.Environ() {
+			split := strings.SplitN(envVar, "=", 2)
+			if len(split) != 2 {
+				continue
+			}
+			key, val := split[0], split[1]
+			if strings.EqualFold(key, "dockerhub_auth_user") {
+				username = val
+			}
+			if strings.EqualFold(key, "dockerhub_auth_password") {
+				secret = val
+			}
+		}
```

Comment:
I'd use `os.LookupEnv` instead of looping the whole environ.

Reply:
I followed the current pattern in our codebase: https://github.com/dagger/dagger/blob/6a2368cf838d33cb1083ee01e29ac89b4846a0c3/plan/task/exec.go#L192-L218. If you prefer, I can change, I don't mind

Reply:
Agreed